### DISCUSSION TITLE
Slower performance with rectangle

### DISCUSSION THREAD

**Discussion by elesto** [rxhanson/Rectangle#746] [General]
- [yes] This is a new bug that hasn't been documented in a closed issue or in the Readme.
macOS version: Monterey
Rectangle version: 0.50

Im wondering how or if rectangle affects performance of the desktop or when moving windows around the desktop. Im running quite an old intel mac and I cant seem to tell if its just placebo or its known that rectangle really does slow things down (a little)

**Comment by rxhanson** [comment]
You could uncheck the box for Snap windows by dragging. Rectangle listens to drag events when that is enabled so that it knows when you reach the edge of the display, and that's the only thing that could have any effect. I would actually be surprised if this did cause any visible difference, though.

**Reply by elesto** [reply]
Thanks will do

**Comment by lacek** [comment]
I've got performance issue too, but with Rectangle itself instead.

After issuing a command (either via menu or shortcut key combinations) it takes more than 5 seconds for the resizing to take effect:

![rectangle_slow_log](https://user-images.githubusercontent.com/1272100/180377465-f163fb38-7958-4497-a06c-30ad7ac84231.gif)

Log:

```
2022-07-22T14:30:13+08:00: AX sizing proposed: (756.0, 872.0), result: (756.0, 871.0)
2022-07-22T14:30:13+08:00: AX position proposed: (0.0, 38.0), result: (-0.0, 38.0)
2022-07-22T14:30:13+08:00: AX sizing proposed: (756.0, 872.0), result: (756.0, 871.0)
2022-07-22T14:30:13+08:00: leftHalf | display: (0.0, 72.0, 1512.0, 872.0), calculatedRect: (0.0, 38.0, 756.0, 872.0), resultRect: (-0.0, 38.0, 756.0, 871.0), srcScreen: Built-in Retina Display, destScreen: Built-in Retina Display, resultScreen: Built-in Retina Display
```

I have Rectangle version 0.57 (63) on M1 MacBook (OS: 12.2.1 (21D62)).

**Reply by lacek** [reply]
Seems nobody else is having the same issue. I am completely out of idea and it is really affecting my productivity. I ended up switching to https://github.com/peterklijn/hammerspoon-shiftit. It is less user friendly but does what I need.

**Reply by lacek** [reply]
Just tried the same version of Rectangle on the same MacBook with OS upgraded to 13.4 (22F66): issue remains but the delay is shorter (~2-3 seconds) and is not specific to Google Chrome only.

Updated to latest version (0.68 (74)) of Rectangle and tada no more delay.

Briefly checked the change log and commit history and can't find anything related in particular. Anyway, problem is solved for me.

**Comment by yuwilbur** [comment]
I've noticed performance issue on my machine as well.

**Comment by rxhanson** [comment]
Issues with moving/resizing taking a long time typically are not due to Rectangle's performance. The most likely scenario is usually that there is another app that is affecting the move/resize of windows of certain applications (see https://github.com/rxhanson/Rectangle/issues/912 for an example of this with Chrome). Rectangle performs very little logic and is just telling the macOS window (accessibility) API the size/position to set windows to. I'll take a look and see what hammerspoon does, but troubleshooting these types of things can be pretty involved. If anyone hits issues, it's most helpful if you can provide some sort determination if certain applications are causing the issue and which ones they are.

**Comment by yuwilbur** [comment]
Thanks Ryan.

After your comment I started experimenting more and I realized the lag mainly happens when I was snapping Chrome. And after further experimentation, I have learned the following:

Behavior
- Does not lag when there's very few tabs (Lags start when there's a certain number of tabs)

What worked
- Clearing Cookies and other side data + Cached images and files

What DIDN'T work
- Removing extensions didn't seem to do much. I removed a lot of extensions and it didn't seem to affect performance
- Restarting my machine
- Removing GB's worth of unused applications to free up storage space

No idea what cached items caused performance hits, but hope this helps some people. My situation is probably rare.

**Reply by rxhanson** [reply]
Thanks for the additional information! Unfortunately, I'm not aware of anything that could be done Rectangle side to compensate. Even resizing Chrome out of box with no pages loaded, by dragging the corner of the window is laggy. Compare it to just resizing Safari by dragging the corner, and it's a really noticeable difference. For me, this is the way Chrome has always been and I stopped using it a long time ago.
